test/integration/test_bitfinex_market.py

```diff
@@ -0,0 +1,368 @@
+
+        order_id = self.market.sell(trading_pair, amount, OrderType.LIMIT,
+                                    quantize_ask_price)
+        # Wait for order creation event
+        self.run_parallel(self.market_logger.wait_for(SellOrderCreatedEvent))
+
+        # Cancel order. Automatically asserts that order is tracked
+        self.market.cancel(trading_pair, order_id)
+
+        [order_cancelled_event] = self.run_parallel(
+            self.market_logger.wait_for(OrderCancelledEvent))
+
+        self.assertEqual(order_cancelled_event.order_id, order_id)
+
+        # Reset the logs
+        self.market_logger.clear()
+
+    def test_execute_limit_buy(self):
+        trading_pair = "ETHUSD"
+        amount: Decimal = Decimal("0.04")
+        quantized_amount: Decimal = self.market.quantize_order_amount(trading_pair,
+                                                                      amount)
+
+        bid_entries = self.market.order_books[trading_pair].bid_entries()
+
+        most_top_bid = next(bid_entries)
+        bid_price: Decimal = Decimal(most_top_bid.price)
+        quantize_bid_price: Decimal = \
+            self.market.quantize_order_price(trading_pair, bid_price)
+        quantize_bid_price = quantize_bid_price + Decimal("0.2")
+
+        order_id = self.market.buy(trading_pair,
+                                   quantized_amount,
+                                   OrderType.LIMIT,
+                                   quantize_bid_price,
+                                   )
+
+        [order_completed_event] = self.run_parallel(
+            self.market_logger.wait_for(BuyOrderCompletedEvent))
+        order_completed_event: BuyOrderCompletedEvent = order_completed_event
+        trade_events: List[OrderFilledEvent] = [t for t in self.market_logger.event_log
+                                                if isinstance(t, OrderFilledEvent)]
+        base_amount_traded: Decimal = sum(t.amount for t in trade_events)
+        quote_amount_traded: Decimal = sum(t.amount * t.price for t in trade_events)
+
+        self.assertTrue([evt.order_type == OrderType.LIMIT for evt in trade_events])
+        self.assertEqual(order_id, order_completed_event.order_id)
+        self.assertAlmostEqual(quantized_amount,
+                               order_completed_event.base_asset_amount)
+        self.assertEqual("USD", order_completed_event.base_asset)
+        self.assertEqual("ETH", order_completed_event.quote_asset)
+        self.assertAlmostEqual(base_amount_traded,
+                               order_completed_event.base_asset_amount)
+        self.assertAlmostEqual(quote_amount_traded,
+                               order_completed_event.quote_asset_amount)
+        self.assertTrue(any([isinstance(event, BuyOrderCreatedEvent) and event.order_id == order_id
+                             for event in self.market_logger.event_log]))
+        # Reset the logs
+        self.market_logger.clear()
```

Comment:
Essentially this is `test_market_buy` and `test_market_sell`. It will be great if you could follow the naming convention. Also, there seems to be times when the order does not get filled and thus the test would fail. It will be good, if you can manipulate the `quantized_bid/ask_price` more to ensure that this passes.